validation-test/SILOptimizer/rdar137960229.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %target-swift-frontend -c %s -verify
+
+func testFatalError(_ message: @autoclosure () -> String = String()) -> Never {
+  Swift.fatalError()
+}
+
+func test() {
+  testFatalError() // warning: Will never be executed
```

Comment:
Might want to remove the comment for clarity, since we don't actually expect a diagnostic here.